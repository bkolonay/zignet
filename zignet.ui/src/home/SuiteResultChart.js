import React, { Component } from 'react';
import ChartistPieChart from '../common/chartist/ChartistPieChart';
import ListPageLink from './ListPageLink';
import LastRunTimeLabel from './LastRunTimeLabel';
import './css/suiteResultChart.css';

class SuiteResultChart extends Component {

  _getChartData(suiteResult) {
    return {
      series: [
        suiteResult.TotalPassedTests,
        suiteResult.TotalFailedTests
      ]
    };
  }

  render() {

    const suiteResult = this.props.suiteResult;
    const suiteId = suiteResult.SuiteIds[0];
    const grouped = this.props.grouped;
    const totalTests = suiteResult.TotalPassedTests + suiteResult.TotalFailedTests;

    return (
      <div className="col-4">
        <h3 className="text-center">{suiteResult.SuiteName}</h3>
        <ChartistPieChart chartId={suiteId} chartData={this._getChartData(suiteResult)} />
        <p className="text-center chart-label">
          <ListPageLink grouped={grouped} suiteId={suiteId} totalTests={totalTests} />
        </p>
        <LastRunTimeLabel grouped={grouped} suiteEndTime={suiteResult.SuiteEndTime} />
      </div>
    );
  }
}

export default SuiteResultChart;
