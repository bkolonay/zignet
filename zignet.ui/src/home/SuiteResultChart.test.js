import React from 'react';
import ReactDOM from 'react-dom';

import SuiteResultChart from './SuiteResultChart';
import renderer from 'react-test-renderer';

it('renders with passed and failed tests', () => {
  const suiteResult = {
    'SuiteID':123456789,
    'SuiteName':'suite-name',
    'TotalPassedTests':12345,
    'TotalFailedTests':678910
  };

  const div = document.createElement('div');
  ReactDOM.render(
    <SuiteResultChart key={suiteResult.SuiteID} suiteResult={suiteResult} />,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders with empty data', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <SuiteResultChart suiteResult={{}} />,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('matches the previous snapshot', () => {
  const suiteResult = {
    'SuiteID':123456789,
    'SuiteName':'suite-name',
    'TotalPassedTests':12345,
    'TotalFailedTests':678910
  };

  const tree = renderer
    .create(<SuiteResultChart key={suiteResult.SuiteID} suiteResult={suiteResult} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});