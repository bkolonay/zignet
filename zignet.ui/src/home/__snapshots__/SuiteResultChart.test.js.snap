// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the previous snapshot 1`] = `
<div
  className="col-4"
>
  <h3
    className="text-center"
  >
    suite-name
  </h3>
  <div
    className="ct-chart"
    id="chart123456789"
  />
  <p
    className="text-center"
  >
    Total: 
    691255
  </p>
</div>
`;
